File: backend/src/main/java/com/whiteblue/backend/domain/user/UserService.java
package com.whiteblue.backend.domain.user;

import com.whiteblue.backend.domain.user.DTO.GetUserResponse;
import com.whiteblue.backend.security.oAuth.OAuthUser;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@RequiredArgsConstructor
@Transactional
@Service
public class UserService {
    private final UserRepository userRepository;

    @Transactional(readOnly = true)
    public GetUserResponse findByAuth(OAuthUser oAuthUser) {
        User user = userRepository.findById(oAuthUser.getId())
                                  .orElseThrow();

        return GetUserResponse.builder()
                              .username(user.getUsername())
                              .name(user.getName())
                              .image(user.getImage())
                              .build();
    }
}

File: backend/src/main/kotlin/com/whiteblue/backend/domain/user/UserService.kt
package com.whiteblue.backend.domain.user

import com.whiteblue.backend.domain.user.dto.GetUserResponse
import com.whiteblue.backend.security.oAuth.OAuthUser
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Transactional
@Service
class UserService(private val userRepository: UserRepository) {
    @Transactional(readOnly = true)
    fun findByAuth(oAuthUser: OAuthUser): GetUserResponse {
        return userRepository.findByOAuthUser(oAuthUser)
            .run {
                GetUserResponse(
                    username = username,
                    name = name,
                    image = image
                )
            }
    }
}
